src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/// <summary>
+/// Task timeout helper based on http://blogs.msdn.com/b/pfxteam/archive/2011/11/10/10235834.aspx
+/// </summary>
+namespace System.Threading.Tasks
+{
+    public static class TaskTimeoutExtensions
+    {
+        public static async Task TimeoutAfter(this Task task, int millisecondsTimeout)
+        {
+            if (task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout)))
```

Comment:
For our testing purposes, it's not going to matter.  To make this a bit more robust, you'd want it to be:
```C#
var cts = new CancellationTokenSource();
if (task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout, cts.Token)))
{
    cts.Cancel();
    await task;
...
```
That way, if the task does complete, we cancel the delay task rather than leaving it to eventually fire and cleanup in the background.  This only matters if we expect tons of these, which we don't.